doc/opt_aero.rst

```diff
@@ -93,20 +93,42 @@ We can set up constraints on the geometry with the DVConstraints class, also fou
+The points are projected normal to the chordwise lines connecting the leading and trailing edge points of this grid.
+The grid is defined by providing four corner points (using ``leList`` and ``teList``) and by specifying the number of spanwise and chordwise points (using ``nSpan`` and ``nChord``).
+By default, ``scaled=True`` for ``addVolumeConstraint()`` and ``addThicknessConstraints2D()``, which means that the volume and thicknesses calculated will be scaled by the initial values (i.e., they will be normalized).
+Therefore, ``lower=1.0`` in this example means that the lower limits for these constraints are the initial values (i.e., if ``lower=0.5`` then the lower limits would be half the initial volume and thicknesses).
+
+.. warning:: The ``leList`` and ``teList`` points must lie completely inside the wing.
+
+.. literalinclude:: ../tutorial/opt/aero/aero_opt.py
+    :start-after: # rst dvconVolThick (beg)
+    :end-before: # rst dvconVolThick (end)
+
+For the volume constraint, the volume is computed by adding up the volumes of the prisms that make up the projected grid as illustrated in the following image (only showing a section for clarity).
+For the thickness constraints, the distances between the upper and lower projected points are used, as illustrated in the following image.
+During optimization, these projected points are also moved by the FFD, just like the wing surface, and are used again to calculate the thicknesses and volume for the new designs.
 More information on the options can be found in the `pyGeo docs <https://mdolab-pygeo.readthedocs-hosted.com/>`_ or by looking at the pyGeo source code.
 
-The LeTe constraints (short for Leading edge/Trailing edge) are linear constraints based on the FFD control points.
+.. image:: images/opt_thickness_and_vol_diagram.png
+   :scale: 40
+   :align: center
+
+The LeTe constraints (short for Leading edge/Trailing edge constraints) are linear constraints based on the FFD control points.
 When we have both twist and local shape variables, we want to prevent the local shape variables from creating a shearing twist.
-This is done by constraining that the upper and lower nodes on the leading and trailing edges must move in opposite directions.
+This is done by constraining the upper and lower FFD control points on the leading and trailing edges to move in opposite directions.
+Note that the LeTe constraint is not related to the ``leList`` and ``teList`` points discussed above.
 
 .. literalinclude:: ../tutorial/opt/aero/aero_opt.py
-    :start-after: # rst dvcon (beg)
-    :end-before: # rst dvcon (end)
+    :start-after: # rst dvconLeTe (beg)
+    :end-before: # rst dvconLeTe (end)
+
+When this optimization example is run, a file named ``constraints.dat`` will be output and can be opened with Tecplot to visualize and check these constraints.
```

Comment:
I think this can be clarified a bit. The `writeTecplot` call is done before the design is changed. So the output file will have some form of visual representations of the original points used for the constraints. To get this, users do not need to "run an optimization", they just need to run the script up to that line, which is right after constraints are set up. 

I am not sure, but I would guess that calling `writeTecplot` would output the updated positions of these points after a design change. Again, I am not sure if this would work, and we may need to test it. 

Finally, these constraint computations are duplicated on all processors, so users should run the `writeTecplot` function on a single processor, as we do in the script. Not sure if this is worth mentioning.

Reply:
Reworded and mentioned that this can also be added at the end to see the final constraints. 
+ Put a comment in the code for the single proc.